homeassistant/components/stream/core.py

```diff
@@ -3,31 +3,53 @@
 
 import asyncio
 from collections import deque
+from collections.abc import Generator
 import datetime
 from typing import TYPE_CHECKING
 
 from aiohttp import web
+import async_timeout
 import attr
 
 from homeassistant.components.http.view import HomeAssistantView
 from homeassistant.core import CALLBACK_TYPE, HomeAssistant, callback
 from homeassistant.helpers.event import async_call_later
 from homeassistant.util.decorator import Registry
 
-from .const import ATTR_STREAMS, DOMAIN, TARGET_SEGMENT_DURATION
+from .const import (
+    ATTR_STREAMS,
+    DOMAIN,
+    SEGMENT_DURATION_ADJUSTER,
+    TARGET_SEGMENT_DURATION_NON_LL_HLS,
+)
 
 if TYPE_CHECKING:
     from . import Stream
 
 PROVIDERS = Registry()
 
 
+class StreamConstants:
+    """Constants that are initialized when Stream is loaded."""
+
+    LL_HLS = False
```

Comment:
Where does this become set to True ever?

I'm having trouble reasoning about constants that are initialized at runtime (from the class description). Are these more like configuration options than constants?

Reply:
Yes, they take the configuration options and set the constants based on them. If the configuration options are not specified they use the defaults, which correspond to the non LL-HLS case. Keeping these constants in a class allows us to easily set them once and then access them from the different files. Is there a better way to do this?

Reply:
They get set in `async_setup`

Reply:
I thought about that before, but the problem is it makes access a lot harder - we have to have hass available everywhere that the "constants" are accessed. They're really only set once when Stream is loaded, but if we're really worried about violating the sanctity of constants, maybe we can just make the variables lower case and rename the class.

Reply:
I renamed `StreamConstants` to `StreamSettings` and lowercased the members. See if that's ok with you or if you think it's still better to try the hass.data approach

Reply:
The name `StreamSettings` looks great -- I do thin a hass.data approach seems more consistent with what i've seen in this code base of how to communicate between configuration for integrations and their libraries (similar to communication between the integration and the platforms it loads)

Reply:
Changed `StreamSettings` to be an instantiable class, so now a `StreamSettings` object gets stored/passed in `hass.data`.